src/StacBrowser.vue

```diff
@@ -134,6 +144,20 @@ export default {
         // Load the root catalog data if not available (e.g. after page refresh or external access)
         this.$store.dispatch("load", { url, loadApi: true });
       }
+    },
+    appStateAsParams () {
+      // Force vue-router to read some values as arrays
+      for (const [key, value] of Object.entries(this.$route.query)) {
+        if (Array.isArray(this.appStateAsParams[key]) && !(Array.isArray(value))) {
+          this.$route.query[key] = [value];
+        }
+      }
+      // Prevent needless navigation which vue-router doesn't lke
+      if (JSON.stringify(this.$route.query) === JSON.stringify(this.appStateAsParams)) {
```

Comment:
Yeah there is perhaps an object-equality library that might help with this comparison.

I think this condition stems from calling `router.push()` for these app state changes, which in turn triggers the `router.afterEach` routine, which in turn overwrites the `stateQueryParameters`, so it's basically an endless loop unless we intervene somehow.

Reply:
I see. There are a couple of questions to answer here:

1. Is push the right function? Or should the state not clutter the browser history and we should use replace?
2. How can we distinguish the state changes from the actual page navigation? Or should stateQueryParameters be read-only and we directly populate the URI so that overwriting is not an issue any more? Haven't had the time to deeply think about it yet...

Reply:
I've pushed up a commit that make things a bit more explicit as per suggestion 2 about what's happening. With the new commit there are no navigation errors that I've come across, but it's a bit more verbose in the code.
And in the new commit I've switched to replace.